core/scrollbar.js

```diff
@@ -306,9 +306,9 @@ Blockly.Scrollbar.prototype.resizeHorizontal_ = function(hostMetrics) {
   this.svgKnob_.setAttribute('width', Math.max(0, innerLength));
   this.xCoordinate = hostMetrics.absoluteLeft + 0.5;
   if (this.pair_ && this.workspace_.RTL) {
-    this.xCoordinate += hostMetrics.absoluteLeft +
-        Blockly.Scrollbar.scrollbarThickness;
+    this.xCoordinate += Blockly.Scrollbar.scrollbarThickness;
```

Comment:
I think this was a mistake before, but usually by a small enough number that it didn't matter.

The scrollbar doesn't start at the beginning in RTL, as you noticed.  The point of this offset is to provide the little white square that neither scrollbar will go into, at the corner where they meet.  So it only needs to move over by the thickness of the other scrollbar.  In LTR mode this is accomplished by capping how far it can move over, but here it needs the correct start position to be set.

Previously absoluteLeft has been < 5 so readding it was fine.  No it can be much larger, due to having a vertical toolbox on the left in RTL mode.  Adding it again means that the scrollbar can scroll offscreen to the right but cannot scroll all the way to the left side of the workspace.